### ISSUE TITLE
Create a RPC interface as an optional replacement to tcl

### ISSUE THREAD

**Issue by sequencer** [YosysHQ/yosys#2047]
Currently, if user wanna interact with yosys, the only way is invoke a tcl shell to send command and receive command. The result is hard to parse, and string only without type struct informations.
So I wonder if possible to add a json/protobuf based RPC interface as a optional replacement to tcl which contains a more serializable API for other program to interact with yosys.

**Comment by whitequark** [comment]
Have you seen pyosys?

**Comment by sequencer** [comment]
Yes, I saw that, but I think that API is a python wrapper to tcl, which still doesn't have a standard API which embedded with Json/Protobuf or something else. Not sure if I'm right?

**Comment by whitequark** [comment]
Nope. Unlike vendor toolchains Yosys isn't built around Tcl, and frankly the Tcl interface is an afterthought.

**Comment by whitequark** [comment]
Closing since pyosys already implements this.
